src/types/input_device.zig

```diff
@@ -49,4 +49,11 @@ pub const InputDevice = extern struct {
 
     extern fn wlr_input_device_get_virtual_keyboard(wlr_dev: *InputDevice) ?*wlr.VirtualKeyboardV1;
     pub const getVirtualKeyboard = wlr_input_device_get_virtual_keyboard;
+
+    extern fn wlr_input_device_is_libinput(wlr_dev: *InputDevice) bool;
+    extern fn wlr_libinput_get_device_handle(wlr_dev: *InputDevice) *opaque {};
+    pub fn getLibinputDevice(wlr_dev: *InputDevice) ?*opaque {} {
```

Comment:
This isn't going to work without a `@ptrCast()` since each `opapue {}` is a distinct type. Probably cleaner to use a constant
```
const LibinputDevice = opaque {};
```

Reply:
Huh, interesting. I need to read up on that a bit more.

Reply:
it's the same principle that makes `struct {} != struct {}` true and that causes e.g. the following structs to be distinct.
```zig
const Foo = struct { x: i32, y: i32 };
const Bar = struct { x: i32, y: i32 };
```
that is, passing a struct of type `Bar` to a `fn foo(f: Foo) void` would be a compile error.